lab_1_classify_by_unigrams/main.py

```diff
@@ -34,6 +44,12 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not (isinstance(tokens, list) and all(isinstance(t, str) for t in tokens)):
+        return None
+    frequencies = {}
+    for l in tokens:
+        frequencies[l] = tokens.count(l)/len(tokens)
```

Comment:
Use `list.count` is a very good idea. But you call it for every symbol (probably the same symbol repeated multiple times). Can you optimize this?

Reply:
ok, good move. Try to use set